lib/api.js

```diff
@@ -27,7 +27,7 @@ function merge (a, b, mapper) {
       !(b[k] instanceof Uint8Array) &&
       !Array.isArray(b[k])
     ) {
-      a[k] = {}
+      a[k] ??= {}
       merge(a[k], b[k], mapper)
     } else {
       a[k] = mapper(b[k], k)
```

Comment:
This is the line I was pointing to - see all the merges end up here and they all do nothing with a[k]... so you ??= above makes no difference?

Reply:
No, the `a[k] ??=` makes a difference and it solved a problem in one of the new tests.

It's about merging objects. If a has `{foo:{x:10}}` and b has `{foo:{y:20}}` then the result will be `{foo:{x:10,y:20}}`. Note that if b had x, then b's x would ovewrite a's x.

Without the `??=` the result would be `{foo:{y:20}}`.